### PR TITLE
feat: add Amazon SageMaker backend dokumentation

### PR DESCRIPTION
**Pull request by dkuroczk**
This PR adds documentation about Amazon SageMaker backend with example how to deploy Amazon SageMaker and how to configure backend.

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

**Review by AlexsJones — APPROVED** [review]

File: docs/reference/providers/backend.md [hunk 1]
```diff
@@ -55,9 +56,26 @@ LocalAI is a local model, which is an OpenAI compatible API. It uses llama.cpp a
...
+
+Example how to deploy Amazon SageMaker with cdk is availabel in [llm-sagemaker-jumpstart-cdk](https://github.com/zaremb/llm-sagemaker-jumpstart-cdk) repo.
+
+- Run the following command to add SageMaker:
+
+```bash
+    k8sgpt auth add --backend amazonsagemaker --providerRegion eu-west-1 --endpointname endpoint-xxxxxxxxxx
+```
+
+- Now you are ready to analyze with the Azure OpenAI backend:
```
Comment (AlexsJones) [inline docs/reference/providers/backend.md:73]: This should be sagemaker?

File: docs/reference/providers/backend.md [hunk 2]
```diff
@@ -55,9 +56,26 @@ LocalAI is a local model, which is an OpenAI compatible API. It uses llama.cpp a
...
     ```
     k8sgpt analyze --explain --backend localai
-
     ```
 
+## Amazon SageMaker
+
+The Amazon SageMaker backend allows you to leverage a self-deployed and managed Language Models (LLM) on Amazon SageMaker.
+
+Example how to deploy Amazon SageMaker with cdk is availabel in [llm-sagemaker-jumpstart-cdk](https://github.com/zaremb/llm-sagemaker-jumpstart-cdk) repo.
```
Comment (AlexsJones) [inline docs/reference/providers/backend.md:65]: Spelling of "availabel" is available

**Review by AlexsJones — CHANGES_REQUESTED** [review]

File: docs/reference/providers/backend.md (hunk 1 above)
Comment (dkuroczk) [reply docs/reference/providers/backend.md]: 🤦 Fixed

File: docs/reference/providers/backend.md (hunk 2 above)
Comment (dkuroczk) [reply docs/reference/providers/backend.md]: done

**Review by AlexsJones — APPROVED** [review]

**Comment by dkuroczk** [comment]
Commit signature fixed ready to be merged

**Review by AlexsJones — APPROVED** [review]

**Pull request merged**